packages/core/useFetch/index.ts

```diff
@@ -0,0 +1,161 @@
+
+  /**
+   * The fetch response body, may either be JSON or text
+   */
+  data: Ref<object | string | null>
+
+  /**
+   * Indicates if the request is currently being fetched.
+   */
+  isFetching: Ref<boolean>
+
+  /**
+   * Indicates if the fetch request is able to be aborted
+   */
+  canAbort: ComputedRef<boolean>
+
+  /**
+   * Abort the fetch request
+   */
+  abort: Fn
+
+  /**
+   * Manually call the fetch
+   */
+  execute: Fn
+}
+
+interface UseFetchOptions {
+  /**
+   * Will automatically run fetch when `useFetch` is used
+   *
+   * @default true
+   */
+  autoFetch?: boolean
+
+  /**
+   * Will automatically refetch when the URL is changed if the url is a ref
+   *
+   * @default false
+   */
+  autoRefetch?: MaybeRef<boolean>
+}
+
+export function useFetch(url: MaybeRef<string>): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, useFetchOptions: UseFetchOptions): UseFetchReturn
+export function useFetch(url: MaybeRef<string>, options: RequestInit, useFetchOptions?: UseFetchOptions): UseFetchReturn
+
+export function useFetch(url: MaybeRef<string>, ...args: any[]): UseFetchReturn {
+  const supportsAbort = typeof AbortController === 'function'
+
+  let fetchOptions: RequestInit = {}
+  let options: UseFetchOptions = { autoFetch: true, autoRefetch: false }
+
+  if (args[0]) {
+    if ('autoFetch' in args[0] || 'autoRefetch' in args[0])
+      options = { ...options, ...args[0] }
+    else
+      fetchOptions = args[0]
+  }
+
+  if (args[1]) {
+    if ('autoFetch' in args[1] || 'autoRefetch' in args[1])
+      options = { ...options, ...args[1] }
+  }
+
+  const isFinished = ref(false)
+  const isFetching = ref(false)
+  const status = ref<number | null>(null)
+  const response = ref<Response | null>(null)
+  const error = ref<any>(null)
+  const data = ref<string | object | null>(null)
+
+  const canAbort = computed(() => {
+    return supportsAbort && isFetching.value
+  })
+
+  let controller: AbortController | null
+
+  const abort = () => {
+    if (supportsAbort && controller instanceof AbortController)
+      controller.abort()
+  }
+
+  const execute = () => {
+    isFetching.value = true
+    isFinished.value = false
+    error.value = null
+
+    if (supportsAbort) {
+      controller = new AbortController()
+      fetchOptions = { ...fetchOptions, signal: controller.signal }
+    }
+
+    fetch(unref(url), fetchOptions)
+      .then((fetchResponse) => {
+        response.value = fetchResponse
+        status.value = fetchResponse.status
+
+        const contentType = fetchResponse.headers.get('content-type')
+
+        if (contentType && contentType.includes('application/json'))
+          fetchResponse.json().then(json => data.value = json)
+        else
+          fetchResponse.text().then(text => data.value = text)
+
+        // see: https://www.tjvantoll.com/2015/09/13/fetch-and-errors/
+        if (!fetchResponse.ok)
+          throw new Error(fetchResponse.statusText || 'Something went wrong!')
```

Comment:
```suggestion
          throw new Error(fetchResponse.statusText)
```

Would like to avoid text strings in VueUse.